src/herder/HerderSCPDriver.cpp

```diff
@@ -697,11 +700,20 @@ HerderSCPDriver::combineCandidates(uint64_t slotIndex,
 
     // take the txSet with the biggest size, highest xored hash that we have
     TxSetFramePtr bestTxSet;
+    std::vector<TimePoint> candidateCloseTimes;
```

Comment:
Why not change `candidateValues` to be a `vector<StellarValue, ValueWrapperPtrSet::iterator>` ?
Then you don't need to track all those things separately, and when picking the best tx set, the index is an iterator in that vector (instead of using optional indexes, etc).

Reply:
I think I've found something even simpler:  just track the maximum `closeTime` _and_ the `closeTime` and signature associated with the best tx set.  Then at the end you just choose once (based on the protocol version) whether it's the max or the best-associated `closeTime` that you take.  That way, there's no need for any new vectors at all, or any iterators or indexes.